distribution/helm/vector-shared/templates/_metrics.tpl

```diff
@@ -39,17 +40,22 @@ Internal metrics are common, so we share and reuse the definition.
 {{- end }}
 {{- end }}
 
+{{- end }}
 {{- end }}
 
 {{/*
 Common Vector container ports used by the built-in metrics pipeline.
 Internal metrics are common, so we share and reuse the definition.
 */}}
 {{- define "libvector.metricsContainerPorts" -}}
-{{- with .Values.prometheusSink }}
```

Comment:
Any specific reason to remove `with` here? It seems to me it tidies up the value queries a bit?

Reply:
Appeared to break the template when referring to `.Values.customConfig`

Reply:
You need to use `$` to reference to the root object in that case (e.g. `$.Values.customConfig`). See the [documentation](https://helm.sh/docs/chart_template_guide/control_structures/#modifying-scope-using-with)

Reply:
🤦